File: typeorm/new-hire-onboard-self-help-portal/src/entity/UserChecklist.ts
import {Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn} from "typeorm";
import { Checklist } from "./Checklist";
import { User } from "./User";

@Entity({name: "user_checklists"})
export class UserChecklist {

    @PrimaryGeneratedColumn()
    id: number;

    @ManyToOne(type => Checklist, checklist => checklist.userChecklists)
    @JoinColumn({name: "checklist_id"})
    checklist: Checklist

    @ManyToOne(type => User, user => user.userChecklists)
    @JoinColumn({name: "user_id"})
    user: User

    @Column()
    assigned_date: string;

    @Column()
    completed_date: string;

    @Column()
    status: boolean;

}

File: spring/new-hire-onboard-self-help-portal/src/main/java/com/example/newhireonboardselfhelpportal/entities/UserChecklist.java
package com.example.newhireonboardselfhelpportal.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonBackReference;

/**
 * UserChecklist
 */
@Entity(name = "user_checklists")
public class UserChecklist {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;

    @JsonBackReference
    @ManyToOne
    @JoinColumn(name="checklist_id")
    private Checklist checklist;

    @JsonBackReference
    @ManyToOne
    @JoinColumn(name="user_id")
    private User user;

    @Column(name = "assigned_date")
    private String assignedDate;

    @Column(name = "completed_date")
    private String completedDate;

    @Column(name = "status")
    private boolean status;

    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Checklist getChecklist() {
        return this.checklist;
    }

    public void setChecklist(Checklist checklist) {
        this.checklist = checklist;
    }

    public User getUser() {
        return this.user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getAssignedDate() {
        return this.assignedDate;
    }

    public void setAssignedDate(String assignedDate) {
        this.assignedDate = assignedDate;
    }

    public String getCompletedDate() {
        return this.completedDate;
    }

    public void setCompletedDate(String completedDate) {
        this.completedDate = completedDate;
    }

    public boolean isStatus() {
        return this.status;
    }

    public boolean getStatus() {
        return this.status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

}
